### PR TITLE
Feat: Implement first iteration of Tab component

### PR DESCRIPTION
**Pull request by aminpaks**
What to expect:
- Implementation of Tab component
How to test:
- Head over to http://localhost:3000/styleguide/tabs

### CODE REVIEW

File: src/Device/breakpoints.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
-  phoneOnly = '599px',
-  tabletPortraitUp = '600px',
-  tabletLandscapeUp = '900px',
-  desktopUp = '1200px',
-  bigDesktopUp = '1800px',
+  phoneOnly = 599,
+  tabletPortrait = 600,
+  tabletLandscape = 900,
+  desktop = 1200,
+  bigDesktop = 1800,
```
Comment (aminpaks) [inline src/Device/breakpoints.ts]: This way we can apply maths operations

File: src/Device/device.util.ts [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
-  tabletPortraitUp: `@media (min-width: ${breakpoints.tabletPortraitUp})`,
-  tabletLandscapeUp: `@media (min-width: ${breakpoints.tabletLandscapeUp})`,
-  desktopUp: `@media (min-width: ${breakpoints.desktopUp})`,
-  bigDesktopUp: `@media (min-width: ${breakpoints.bigDesktopUp})`,
+  phoneOnly: `@media (max-width: ${breakpoints.phoneOnly}px)`,
+  tabletPortraitMin: `@media (min-width: ${breakpoints.tabletPortrait}px)`,
+  tabletPortraitMax: `@media (max-width: ${breakpoints.tabletPortrait - 1}px)`,
+  tabletLandscapeMin: `@media (min-width: ${breakpoints.tabletLandscape}px)`,
+  desktopMin: `@media (min-width: ${breakpoints.desktop}px)`,
+  bigDesktopMin: `@media (min-width: ${breakpoints.bigDesktop}px)`,
```
Comment (aminpaks) [inline src/Device/device.util.ts]: Min & max is more handy

File: src/Styleguide/Tabs/Tabs.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    <>
+      <HeadTitle title="Tabs" suffix="Styleguide" />
+      <Section>
+        <StyledSectionContainer>
+          <Tab
+            items={[
+              { id: 'c1', title: 'Tab 1', content: getLorem('Tab 1 content') },
+              { id: 'c2', title: 'Tab 2', content: getLorem('Tab 2 content') },
+              { id: 'c3', title: 'Tab 3', content: getLorem('Tab 3 content') },
+              { id: 'c4', title: 'Tab 4', content: getLorem('Tab 4 content') },
```
Comment (aminpaks) [inline src/Styleguide/Tabs/Tabs.tsx]: I built it with multiple interface and I ended up simplifying it with a data model like this.
It accepts "ReactNode" as content too.

File: src/Device/breakpoints.ts (hunk 1 above)
Comment (nasreddineskandrani) [reply src/Device/breakpoints.ts]: great idea!

File: src/Device/device.util.ts (hunk 2 above)
Comment (nasreddineskandrani) [reply src/Device/device.util.ts]: as discussed on slack, `phoneOnly` is the same as `tabletPortraitMax`
we should remove `tabletPortraitMax` and keep using `phoneOnly`

Comment (aminpaks) [reply src/Device/device.util.ts]: Let's merge it, it's fine. Having more descriptive values is better to refer to other non-relative ones.

Comment (nasreddineskandrani) [reply src/Device/device.util.ts]: tell me what `tabletPortraitMax` describe?

Comment (aminpaks) [reply src/Device/device.util.ts]: It describes the table portrait starts from that value

Comment (nasreddineskandrani) [reply src/Device/device.util.ts]: table portrait starts from that value => `tabletPortraitMin`

Comment (aminpaks) [reply src/Device/device.util.ts]: sorry it ends there

Comment (aminpaks) [reply src/Device/device.util.ts]: so we should switch between min and max then

**Comment by nasreddineskandrani** [comment]
tag #36

**Pull request merged**